pandas/plotting/_matplotlib/core.py

```diff
@@ -400,9 +400,14 @@ def _compute_plot_data(self):
         # GH16953, _convert is needed as fallback, for ``Series``
         # with ``dtype == object``
         data = data._convert(datetime=True, timedelta=True)
-        numeric_data = data.select_dtypes(
-            include=[np.number, "datetime", "datetimetz", "timedelta"]
-        )
+
+        # GH22799, exclude datatime type for boxplot
+        include_type = [np.number, "datetime", "datetimetz", "timedelta"]
+        if self._kind == "box":
+            include_type.remove("datetime")
```

Comment:
Does datetimetz work? Or does it need to be excluded here too?

Reply:
ahh, yeah, should exclude `datetime` and `timedelta`... thanks for pointing out! @TomAugspurger